haystack/schema.py

```diff
@@ -637,7 +678,7 @@ def __init__(self, labels: List[Label], drop_negative_labels=False, drop_no_answ
         # If we do not exclude them from document_ids this would be problematic for retriever evaluation as they do not contain the answer.
         # Hence, we exclude them here as well.
         self._document_ids = [l.document.id for l in self._labels if not l.no_answer]
-        self._contexts = [str(l.document.content) for l in self._labels if not l.no_answer]
+        self._contexts = [l.document.to_dict()["content"] for l in self._labels if not l.no_answer]
```

Comment:
Updated to use `document.to_dict()["content"]` so we can store table contents as a list of lists which is compatible with MultiLabel serialization (example shown [here](https://github.com/deepset-ai/haystack/pull/4616/files#diff-f5415b42f1092e8049cc80f502da84663a183b263fedba622d4dff7ab87d4da3R540)), and still preserves the structure of the table. Before we stringified a pandas dataframe which is non-intuitive to read and use.